libraries/node-core-library/src/Resolve.ts

```diff
@@ -87,7 +95,7 @@ export class ResolveUtilities {
       pathInsidePackage = packagePath.substr(lastSlashIndex + 1);
```

Comment:
[GitHub is deficient, so I have to copy+paste these lines in order to attach a comment.]
```ts
      return path.dirname(
        sync(packageName, {
          basedir: normalizedRootPath,
          packageFilter: (pkg: { main: string }): { main: string } => {
            pkg.main = 'package.json';
            return pkg;
          }
        })
      );
```

This code does something right:
- It patches the `"main"` field to point to `package.json`, so we can find the package folder even in cases where the entry point is in a subdirectory

But it has two other bugs:
- You forgot to add `preserveSymlinks: false`, so it applies obsolete Node.js lookups. The `resolve` package never fixed their default, so everyone has to remember to add this setting, otherwise it resolves incorrectly.
- By default `resolve` will also resolve Node.js built-in libraries. For example, when `package.json` refers to `http`, it means the NPM package of that name. Whereas `resolve` assumes you mean the Node.js `http` module. The workaround is to append a slash, [as we did here](https://github.com/microsoft/rushstack/blob/de6d949aa5cce8c9ed957b552c6e2fe3ba13dd0d/apps/rush-lib/src/logic/deploy/DeployManager.ts#L291)

Generally speaking, the `resolve.sync` API design is just super duper error prone. I've been wanting to banish all references to that API across our code base (of which there are many), and wrap it up behind a tidy `node-core-library` wrapper like `Import.resolveThing()`.  Maybe your PR can blaze that trail?